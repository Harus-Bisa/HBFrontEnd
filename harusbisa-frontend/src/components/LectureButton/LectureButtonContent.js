import React from "react";

export default function LectureButtonContent(props){
    const Icon = props.icon
    return(
        <div>
            <div className="row">
                <div className="col-4" style={{display:'flex'}}>
                    {Icon && <Icon className="student-icon" style={{margin:'auto'}}/>}
                </div>
                <div className="col-8" style={{display:'flex'}}>
                    <p style={{margin:'auto'}}>{props.content}</p>
                </div>
            </div>
        </div>
    )
}