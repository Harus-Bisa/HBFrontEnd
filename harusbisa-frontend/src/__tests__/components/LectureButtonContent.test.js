import React from "react";
import {shallow} from "enzyme";
import '../../test-config';
import LectureButtonContent from "../../components/LectureButton/LectureButtonContent";
import AddIcon from '@material-ui/icons/Add';

describe("LectureButtonContent component", () =>{
    it("matches snapshot", () =>{
        let component = shallow(<LectureButtonContent/>)
        expect(component.html()).toMatchSnapshot();
    })

    it("renders icon", () =>{
        let component = shallow(<LectureButtonContent icon={AddIcon}/>)
        let icon = component.find(".student-icon")
        expect(icon.html()).not.toBe(null)
    })
    it("does not renders icon", () =>{
        let component = shallow(<LectureButtonContent/>)
        let icon = component.find(".student-icon")
        expect(icon).toHaveLength(0)
    })
})