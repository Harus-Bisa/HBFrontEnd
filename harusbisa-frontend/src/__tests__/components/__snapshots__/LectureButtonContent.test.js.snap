// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LectureButtonContent component matches snapshot 1`] = `"<div><div class=\\"row\\"><div class=\\"col-4\\" style=\\"display:flex\\"></div><div class=\\"col-8\\" style=\\"display:flex\\"><p style=\\"margin:auto\\"></p></div></div></div>"`;
